### DISCUSSION TITLE
How to create Style policy via code

### DISCUSSION THREAD

**Discussion by pkhand0** [adobe/aem-core-wcm-components#1450] [Q&A]
Hi Team,

I am trying to enable Style policy in my local but not seeing that in image component.

<img width="619" alt="Screen Shot 2021-03-17 at 5 35 10 PM" src="https://user-images.githubusercontent.com/28389958/111464688-2221cb80-8747-11eb-8051-e9501167395a.png">

Also, can you please let us know, apart from UI how i can add these policy from my code

**Comment by vladbailescu** [comment]
@pkhand0 , the styles are persisted as part of a component's Content Policy, for example: https://github.com/adobe/aem-core-wcm-components/blob/master/examples/ui.content/src/content/jcr_root/conf/core-components-examples/settings/wcm/policies/.content.xml#L371-L381

That policy will need to be mapped to a template, for example: https://github.com/adobe/aem-core-wcm-components/blob/master/examples/ui.content/src/content/jcr_root/conf/core-components-examples/settings/wcm/templates/content-page/policies/.content.xml#L67-L70

**Reply by pkhand0** [reply]
Thanks @vladbailescu  for reply. 

How to create this policy_xxxxx in our code.. I know i can do it via design dialoge but how to create t

I think this is kind of some dynamic number getting generated ?

    <policy_1550289110975
                        jcr:lastModified="{Date}2019-02-16T04:53:01.777+01:00"
                        jcr:lastModifiedBy="admin"
                        jcr:primaryType="nt:unstructured"
                        jcr:title="Core Components Examples - Form Text"
                        sling:resourceType="wcm/core/components/policy/policy">
                        <jcr:content jcr:primaryType="nt:unstructured"/>
                        <cq:styleGroups jcr:primaryType="nt:unstructured">
                            <item0 jcr:primaryType="nt:unstructured">
                                <cq:styles jcr:primaryType="nt:unstructured">
                                    <item0
                                        cq:styleClasses="cmp-examples-form-text--search"
                                        cq:styleId="1550289148555"
                                        cq:styleLabel="Search"
                                        jcr:primaryType="nt:unstructured"/>
                                </cq:styles>
                            </item0>
                        </cq:styleGroups>
                    </policy_1550289110975>

**Reply by vladbailescu** [reply]
That's just an epoch timestamp added to avoid conflicts. Technically, you could have any valid node name.

**Reply by pkhand0** [reply]
Thanks @vladbailescu  . So i can manually add the policy with any random number.

Will try to implement the change and let u know
